### ISSUE TITLE
NF crash with singularity option activated

### ISSUE THREAD

**Issue by Lcornet** [nextflow-io/nextflow#2309]
## Bug report 

Fail of the workflow when singularity option is activated

### Expected behavior and actual behavior

I simplify the problem with a very short Nextflow script. The NF script work fine without singularity but fail when  the container is used.

### Steps to reproduce the problem

This my Nextflow script:

#!/usr/bin/env nextflow

//test
process makeTEST {

    //informations

        //input output
    input:

    output:

    //script
    script:

    """
    echo 'PASS' > log
    """

}

And this is my classical command:

/root/nextflow test.nf
N E X T F L O W  ~  version 21.04.3
Launching `test.nf` [marvelous_faggin] - revision: b1356710d7
executor >  local (1)
[19/bb78b4] process > makeTEST [100%] 1 of 1 ✔

And my singularity command:
/root/nextflow test.nf -with-singularity test.sif

Error executing process > 'makeTEST'

Caused by:
  Process `makeTEST` terminated with an error exit status (1)

Command executed:

  echo 'PASS' > log

Command exit status:
  1

Command output:
  (empty)

Command error:
  /bin/bash: line 0: cd: /data/test/work/b2/bad447f6f689c44cf47ea9586750dd: No such file or directory
  /bin/bash: .command.sh: No such file or directory

Work dir:
  /data/test/work/b2/bad447f6f689c44cf47ea9586750dd

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

### Program output 

Sep-12 15:10:28.042 [main] DEBUG nextflow.cli.Launcher - $> nextflow test.nf
Sep-12 15:10:28.147 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 21.04.3
Sep-12 15:10:28.168 [main] INFO  nextflow.cli.CmdRun - Launching `test.nf` [wise_lalande] - revision: 8a762ffdd0
Sep-12 15:10:28.180 [main] DEBUG nextflow.config.ConfigBuilder - Found config local: /data/test/nextflow.config
Sep-12 15:10:28.181 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /data/test/nextflow.config
Sep-12 15:10:28.198 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `standard`
Sep-12 15:10:28.803 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; plugins-dir=/root/.nextflow/plugins
Sep-12 15:10:28.804 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[]
Sep-12 15:10:28.806 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins local root: .nextflow/plr/empty
Sep-12 15:10:28.812 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Enabled plugins: []
Sep-12 15:10:28.813 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Disabled plugins: []
Sep-12 15:10:28.816 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode
Sep-12 15:10:28.824 [main] INFO  org.pf4j.AbstractPluginManager - No plugins
Sep-12 15:10:28.918 [main] DEBUG nextflow.Session - Session uuid: 6dabac5e-35f3-4f84-857c-35183bcb40a2
Sep-12 15:10:28.918 [main] DEBUG nextflow.Session - Run name: wise_lalande
Sep-12 15:10:28.919 [main] DEBUG nextflow.Session - Executor pool size: 8
Sep-12 15:10:28.952 [main] DEBUG nextflow.cli.CmdRun -
  Version: 21.04.3 build 5560
  Created: 21-07-2021 15:09 UTC (17:09 CEST)
  System: Linux 4.15.0-154-generic
  Runtime: Groovy 3.0.7 on OpenJDK 64-Bit Server VM 11.0.11+9-Ubuntu-0ubuntu2.18.04
  Encoding: UTF-8 (UTF-8)
  Process: 64403@fsc-cloud85.segi.ulg.ac.be [139.165.70.42]
  CPUs: 8 - Mem: 31.4 GB (15.2 GB) - Swap: 69.9 GB (69.9 GB)
Sep-12 15:10:28.973 [main] DEBUG nextflow.Session - Work-dir: /data/test/work [xfs]
Sep-12 15:10:28.974 [main] DEBUG nextflow.Session - Script base path does not exist or is not a directory: /data/test/bin
Sep-12 15:10:28.987 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[]
Sep-12 15:10:29.000 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory
Sep-12 15:10:29.093 [main] DEBUG nextflow.Session - Session start invoked
Sep-12 15:10:29.234 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution
Sep-12 15:10:29.255 [main] DEBUG nextflow.Session - Workflow process names [dsl1]: makeTEST
Sep-12 15:10:29.314 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: null
Sep-12 15:10:29.314 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'local'
Sep-12 15:10:29.319 [main] DEBUG nextflow.executor.Executor - [warm up] executor > local
Sep-12 15:10:29.325 [main] DEBUG n.processor.LocalPollingMonitor - Creating local task monitor for executor 'local' > cpus=
8; memory=31.4 GB; capacity=8; pollInterval=100ms; dumpInterval=5m
Sep-12 15:10:29.465 [main] DEBUG nextflow.script.ScriptRunner - > Await termination
Sep-12 15:10:29.465 [main] DEBUG nextflow.Session - Session await
Sep-12 15:10:29.606 [Task submitter] DEBUG nextflow.executor.LocalTaskHandler - Launch cmd line: /bin/bash -ue .command.run
Sep-12 15:10:29.612 [Task submitter] INFO  nextflow.Session - [8c/6059e4] Submitted process > makeTEST
Sep-12 15:10:29.897 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 1; name: makeTES
T; status: COMPLETED; exit: 1; error: -; workDir: /data/test/work/8c/6059e42cb3aea819277998e69327ef]
Sep-12 15:10:29.925 [Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'makeTEST'

Caused by:
  Process `makeTEST` terminated with an error exit status (1)

Command executed:

  echo 'PASS' > log

Command exit status:
  1

Command output:
  (empty)

Command error:
  /bin/bash: line 0: cd: /data/test/work/8c/6059e42cb3aea819277998e69327ef: No such file or directory
  /bin/bash: .command.sh: No such file or directory

Work dir:
  /data/test/work/8c/6059e42cb3aea819277998e69327ef

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`
Sep-12 15:10:29.935 [Task monitor] DEBUG nextflow.Session - Session aborted -- Cause: Process `makeTEST` terminated with an
 error exit status (1)
Sep-12 15:10:29.936 [main] DEBUG nextflow.Session - Session await > all process finished
Sep-12 15:10:30.000 [main] DEBUG nextflow.Session - Session await > all barriers passed
Sep-12 15:10:30.011 [main] DEBUG nextflow.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=0
; failedCount=1; ignoredCount=0; cachedCount=0; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCo
unt=0; succeedDuration=0ms; failedDuration=243ms; cachedDuration=0ms;loadCpus=0; loadMemory=0; peakRunning=1; peakCpus=1; p
eakMemory=0; ]
Sep-12 15:10:30.142 [main] DEBUG nextflow.CacheDB - Closing CacheDB done
Sep-12 15:10:30.154 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye

### Environment 

* Nextflow version: 21.04.3
* Java version: openjdk 11.0.11 2021-04-20
* Operating system: Linux
* Bash version: GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)

### Additional context

The same problem appear if I use nextflow.config:

process.container = 'file:///data/test/test.sif'
singularity.enabled = true
singularity.cacheDir = "$PWD"

**Comment by pditommaso** [comment]
This may happen if your Singularity installation is not mounting the /data/test file system by default. Try using the `singularity.autoMounts` option. See https://www.nextflow.io/docs/latest/singularity.html

**Comment by Lcornet** [comment]
The singularity.autoMounts din't help. My problem was due to a symlink in the scratch or our HPC. This config make it work:

process.container = '/scratch/ulg/bioec/lcornet/ORPER/ORPER.sif'
singularity.enabled = true
singularity.cacheDir = "$PWD"
singularity.autoMounts = false
singularity.runOptions = '-B <PATH to dir> -B <path to symlink>'

Nevertheless, I noticed that Nextflow write a lot of temporary files in the /tmp of the cluster. How can I force Nextflow to write temporary files in directory and not in /tmp ? (I thought it was linked to singularity.cacheDir = "$PWD", but no).
